mantidproject/mantid
# Muon Analysis 2 Crash When Fitting

**Original reporter: James**

### Problem
When adding ExpDecayOsc and a third order polynomial to Muon data Mantid crashes.

**Description of work.**

When an attribute was changed in the fitting view all the previous functions were discarded. Fixed by modifying the emit signal to return the complete function.

Fixes #26788

### To Test

    Open Muon Analysis 2 GUI
    Load Current Run
    In the fitting tab, right click and add function
    Under Muon, add ExpDecayOsc function
    Under Background add Polynomial
    Under Polynomial change n to 3
    Press Fit
    Should no longer crash 
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
